components/TemplateGrid/TemplateGridContainer.vue

```diff
@@ -249,42 +189,58 @@ export default {
                 const parent = Math.floor(Math.max(...findElements.map(item => item.row)));
                 parentId = this.dataWithoutGhostElement[parent].id;
             }
-            if (this.dataWithoutGhostElement.length >= this.rows - 2) {
-                this.$emit('setRowsCount', { key: 'rowsCount', value: this.rows + 1 });
+            return parentId;
+        },
+        setGhostItemPosition({ column, row }) {
+            if (
+                (this.ghostElement.column !== column || this.ghostElement.row !== row)
+                && row !== null
+                && column !== null
+            ) {
+                this.ghostElement.row = row;
+                this.ghostElement.column = column;
+                this.addItem({ ...this.ghostElement });
             }
-            this.$emit('addItem', {
-                item: {
-                    id: this.draggedElement,
-                    column,
-                    row,
-                    parent: parentId,
-                },
-            });
-            this.changeItemRelation({ row, column, parentId: this.draggedElement });
-            this.$emit('rebuildGrid', { id: this.draggedElement });
-            return 1;
         },
-        changeItemRelation({ row, column, parentId }) {
-            const findBrothers = this.dataWithoutGhostElement.filter(
-                e => (e.column === column && e.row < row),
-            );
-            if (findBrothers.length) {
-                const brother = Math.floor(Math.max(...findBrothers.map(item => item.row)));
-                const brotherId = this.dataWithoutGhostElement[brother].id;
-                const childrenToChange = this.dataWithoutGhostElement.filter(
-                    e => (e.parent === brotherId && e.row > row),
-                );
-                childrenToChange.forEach((e) => {
-                    this.$emit('addItem', {
-                        item: {
-                            id: e.id,
-                            column: e.column,
-                            row: e.row,
-                            parent: parentId,
-                        },
-                    });
-                });
+        getMouseOverProps(clientX, clientY) {
+            const elements = document.elementsFromPoint(clientX, clientY);
+            const shadowItem = elements.find(e => e.classList.contains('shadow-grid-item'));
```

Comment:
shadow item ? ghost element ?

Reply:
Shadow-item is the element responsible for displaying the logic of grid presentation and has nothing to do with ghost-eleemnt